synapse/rest/admin/rooms.py

```diff
@@ -253,6 +254,56 @@ async def on_GET(self, request):
         return 200, response
 
 
+class RoomRestServlet(RestServlet):
+    """Get room details.
+
+    TODO: Add on_POST to allow room creation without joining the room
+    """
+
+    PATTERNS = admin_patterns("/rooms/(?P<room_id>[^/]+)$")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        ret = await self.store.get_room_with_stats(room_id)
+        if not ret:
+            raise NotFoundError("Room not found")
+
+        return 200, ret
+
+
+class RoomMembersRestServlet(RestServlet):
+    """Get room members as a list of user IDs.
+
+    TODO: Return power level for each user
+    TODO: Add on_POST to allow member invitations
+    """
+
+    PATTERNS = admin_patterns("/rooms/(?P<room_id>[^/]+)/members$")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        start = parse_integer(request, "from", default=0)
+        limit = parse_integer(request, "limit", default=100)
+
+        members = await self.store.get_users_in_room(room_id)
+        if not members:
+            raise NotFoundError("Room not found")
+
+        return 200, { "members": members, "total": len(members) }
```

Comment:
Did you meant to add this in your last commit?

Reply:
No, sorry. Have to get more concentrated with this.
Did a force push to revert this one.